File: src/main/java/dev/canon/cue/module/modules/function/BoatPlace.java
package dev.canon.cue.module.modules.function;

import dev.canon.cue.event.events.client.PacketEvent;
import dev.canon.cue.module.Category;
import dev.canon.cue.module.Module;
import dev.canon.cue.module.ModuleInfo;
import dev.canon.cue.module.Category;
import dev.canon.cue.module.Module;
import dev.canon.cue.module.ModuleInfo;
import net.minecraft.init.Items;
import net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

@ModuleInfo(name = "BoatPlace",
        descriptions = "BoatPlace",
        category = Category.FUNCTION)
public class BoatPlace extends Module {
    @SubscribeEvent
    public void onPacketSend(PacketEvent.Send event) {
        if (event.getPacket() instanceof CPacketPlayerTryUseItemOnBlock) {
            if (mc.player.getHeldItemMainhand().getItem() == Items.BOAT) {
                event.setCanceled(true);
            }
        }
    }
}

File: src/main/java/cn/origin/cube/module/modules/function/BoatPlace.kt
package cn.origin.cube.module.modules.function

import cn.origin.cube.core.events.client.PacketEvent
import cn.origin.cube.core.module.Category
import cn.origin.cube.core.module.Module
import cn.origin.cube.core.module.interfaces.Constant
import cn.origin.cube.core.module.interfaces.ModuleInfo
import net.minecraft.init.Items
import net.minecraft.network.Packet
import net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Constant(constant = false)
@ModuleInfo(name = "BoatPlace", descriptions = "BoatPlace", category = Category.FUNCTION)
class BoatPlace: Module() {
    @SubscribeEvent
    fun onPacketSend(event: PacketEvent.Send) {
        if (event.getPacket<Packet<*>>() is CPacketPlayerTryUseItemOnBlock) {
            if (mc.player.heldItemMainhand.item === Items.BOAT) {
                event.isCanceled = true
            }
        }
    }
}